src/Microsoft.AspNet.Mvc.ViewFeatures/Resources.resx

```diff
@@ -122,13 +122,13 @@
     <comment>{1} is the newline character</comment>
   </data>
   <data name="ViewComponent_AsyncMethod_ShouldReturnTask" xml:space="preserve">
-    <value>The async view component method '{0}' should be declared to return Task&lt;T&gt;.</value>
+    <value>Method '{0}' of view component '{1}' should be declared to return {2}&lt;T&gt;.</value>
   </data>
   <data name="ViewComponent_MustReturnValue" xml:space="preserve">
     <value>A view component must return a non-null value.</value>
   </data>
   <data name="ViewComponent_SyncMethod_ShouldReturnValue" xml:space="preserve">
-    <value>The view component method '{0}' should be declared to return a value.</value>
+    <value>Method '{0}' of view component '{1}' should be declared to return a value.</value>
```

Comment:
Could add a comment saying an alternative (if you don't want to return a value) is to inject services.